app/jenkins-for-jira-ui/src/components/ConnectJenkins/ConnectJenkins/ConnectJenkins.tsx

```diff
@@ -35,9 +39,15 @@ const ConnectJenkins = () => {
 	}, [uuid]);
 
 	useEffect(() => {
+		const analyticsClient = new AnalyticsClient();
+		analyticsClient.sendAnalytics(
+			AnalyticsEventTypes.ScreenEvent,
+			AnalyticsScreenEventsEnum.ConnectJenkinsServerScreenName,
+			{ jiraHost }
```

Comment:
jiraHost can be set from within the `AnalyticsClient` instead of passing it when sending events. 
I'd suggest to get the `siteUrl` within the `AnalyticsClient`, which can be fetched within the `view.getContext()`.

https://developer.atlassian.com/platform/forge/custom-ui-bridge/view/#returns

Reply:
You can't do that with useEffect. The object construction of analyticsClient makes the dependencies in the useEffect arrat change on every render so it has to be defined like this so it doesn't need to be included in the array and cause side effects.